make run-nexus slow or fails on M1

Using the same context as in #360 make run-nexus first succeeded but it took a long time. When aiming to measure this in more detail nexus startup failed.

Counting the waiting dots nexus startup took
- 70 seconds on a beefy Intel Mac.
- 250 seconds on the M1

Then I added some date statements into the bash script that starts nexus as I also had a feeling that it behaved slow on the M1 after startup. However the next attempt to run the image now failed. Here is an excerpt:

```
→ make run-nexus
cd scripts && ./run-nexus.sh
Run container using image tag 3.30.1
Error: No such container: ods-test-nexus
[+] Building 1.4s (6/6) FINISHED                                                                                                                                                                                                          
.. => => naming to docker.io/library/ods-test-nexus                                                                                                                                                                                    0.0s
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
81c38fb9761833690a56039755290cd9f69c1a0d3e30ed5fb3576171f7aceca2
...
Thu Dec 30 12:00:37 CET 2021
Waiting up to 5 minutes for Nexus to start ...
..............................Nexus did not start, got http_code=000.
java.io.IOException: Function not implemented
	at sun.nio.fs.LinuxWatchService.<init>(LinuxWatchService.java:64)
	at sun.nio.fs.LinuxFileSystem.newWatchService(LinuxFileSystem.java:47)
	at com.install4j.runtime.launcher.FullLauncherHelper.watchDirectory(FullLauncherHelper.java:52)
	at com.install4j.runtime.launcher.util.SingleInstance.createStartupListener(SingleInstance.java:108)
	at com.install4j.runtime.launcher.util.SingleInstance.check(SingleInstance.java:95)
	at com.install4j.runtime.launcher.util.SingleInstance.checkForCurrentLauncher(SingleInstance.java:31)
	at com.install4j.runtime.launcher.UnixLauncher.checkSingleInstance(UnixLauncher.java:88)
	at com.install4j.runtime.launcher.UnixLauncher.main(UnixLauncher.java:67)
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGILL (0x4) at pc=0x000000400ceb3f05, pid=1, tid=0x00000040c63ef700
#
# JRE version: OpenJDK Runtime Environment (8.0_292-b10) (build 1.8.0_292-b10)
# Java VM: OpenJDK 64-Bit Server VM (25.292-b10 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# J 134 C1 java.util.zip.ZipFile.ensureOpen()V (37 bytes) @ 0x000000400ceb3f05 [0x000000400ceb3ee0+0x25]
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /opt/sonatype/nexus/hs_err_pid1.log
Compiled method (c1)    5736 1283       3       java.util.zip.ZipFile$ZipEntryIterator::hasMoreElements (5 bytes)
 total in heap  [0x000000400d1daed0,0x000000400d1db258] = 904
 relocation     [0x000000400d1daff8,0x000000400d1db030] = 56
 main code      [0x000000400d1db040,0x000000400d1db140] = 256
 stub code      [0x000000400d1db140,0x000000400d1db1e8] = 168
 metadata       [0x000000400d1db1e8,0x000000400d1db1f8] = 16
 scopes data    [0x000000400d1db1f8,0x000000400d1db210] = 24
 scopes pcs     [0x000000400d1db210,0x000000400d1db250] = 64
 dependencies   [0x000000400d1db250,0x000000400d1db258] = 8
#
# If you would like to submit a bug report, please visit:
#   https://bugzilla.redhat.com/enter_bug.cgi?product=Red%20Hat%20Enterprise%20Linux%208&component=java-1.8.0-openjdk
#
qemu: uncaught target signal 6 (Aborted) - core dumped
Thu Dec 30 12:05:38 CET 2021
make: *** [run-nexus] Error 1
```

Attached are the container logs for the first and second run:
-  [run-nexus-2.log](https://github.com/opendevstack/ods-pipeline/files/7792654/run-nexus-2.log)
- [run-nexus-3.log](https://github.com/opendevstack/ods-pipeline/files/7792655/run-nexus-3.log)

--- a/scripts/nexus/Dockerfile.arm64
+++ b/scripts/nexus/Dockerfile.arm64
@@ -1,0 +1,6 @@
+FROM klo2k/nexus3:3.30.1-01
+
+USER root
+RUN echo "storage.diskCache.diskFreeSpaceLimit=2048" >> /opt/sonatype/nexus/etc/karaf/system.properties && \
+    echo "nexus.scripts.allowCreation=true" >> /opt/sonatype/nexus/etc/nexus-default.properties
+USER nexus

--- a/scripts/run-nexus.sh
+++ b/scripts/run-nexus.sh
@@ -29,7 +29,11 @@ esac; shift; done
 echo "Run container using image tag ${NEXUS_IMAGE_TAG}"
 docker rm -f ${CONTAINER_NAME} || true
 cd "${SCRIPT_DIR}"/nexus
-docker build -t ${IMAGE_NAME} .
+if [ "$(uname -m)" == "arm64" ]; then
+    docker build -t ${IMAGE_NAME} -f Dockerfile.arm64 .
+else 
+    docker build -t ${IMAGE_NAME} .
+fi 
 cd - &> /dev/null
 docker run -d -p "${HOST_PORT}:8081" --net kind --name ${CONTAINER_NAME} ${IMAGE_NAME}
 

